tests/terminator_tests/improvement_tests/test_evaluator.py

```diff
@@ -60,11 +61,25 @@ def test_evaluate() -> None:
         assert regret_bound == 2.0 * np.sqrt(_get_beta(n_params=1, n_trials=len(trials)))
 
 
+def test_best_value_stagnation_evaluate() -> None:
+    evaluator = BestValueStagnationEvaluator(max_stagnation_trials=1)
+    trials = [create_trial(value=value) for value in [0, 1, 2]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == 1
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
+    trials = [create_trial(value=value) for value in [0, 1, 0]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == 0
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
+    trials = [create_trial(value=value) for value in [0, 0, 0]]
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MAXIMIZE) == -1
+    assert evaluator.evaluate(trials=trials, study_direction=StudyDirection.MINIMIZE) == -1
+
+
 def test_evaluate_with_no_trial() -> None:
-    evaluator = RegretBoundEvaluator()
+    evaluators = [RegretBoundEvaluator(), BestValueStagnationEvaluator()]
 
     with pytest.raises(ValueError):
-        evaluator.evaluate(trials=[], study_direction=StudyDirection.MAXIMIZE)
+        for evaluator in evaluators:
+            evaluator.evaluate(trials=[], study_direction=StudyDirection.MAXIMIZE)
```

Comment:
Following the convention of this codebase, we want to use `@pytest.mark.parametrize`, for example, as follows:


```python
@pytest.mark.parametrize(
    "evaluator", [RegretBoundEvaluator(), BestValueStagnationEvaluator()])
)
def test_evaluate_with_no_trial(evaluator: BaseImprovementEvaluator) -> None:

    with pytest.raises(ValueError):
        evaluator.evaluate(trials=[], study_direction=StudyDirection.MAXIMIZE)
```

Reply:
Done